src/Dialog/DialogRouter.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { getDialogType, getDialogProps, getDialogIsOpen } from './selectors';
+import { closeDialog } from './actions';
+import * as DIALOGTYPES from './types';
+import DownloadObjectDialog from '../List/DownloadObjectDialog';
+import ColumnConfigDialog from '../List/columns/ColumnConfigDialog';
+
+const DialogComponents = {
+    [DIALOGTYPES.DOWNLOAD_OBJECT]: DownloadObjectDialog,
+    [DIALOGTYPES.COLUMN_CONFIG]: ColumnConfigDialog,
+};
+
+export class DialogRouter extends Component {
+    state = {
+        openDialog: null,
+    };
+
+    render() {
+        const { open, dialogType, dialogProps, ...rest } = this.props;
+        if (!open) {
+            return null;
+        }
+        const DialogType = DialogComponents[dialogType];
+        return DialogType ? (
+            <DialogType {...{ ...rest, ...dialogProps }} open={true} />
```

Comment:
I was just looking through this PR to learn a bit about the maintenance-app codebase, and was wondering, could this line not be replaced by:

```
<DialogType { ...rest, ...dialogProps } open={true} />
```

Or am I overlooking something obvious?

Reply:
Ah, I assume it's because the jsx spread syntax only allows you to spread a single object.

Reply:
It's kinda ugly, but its a tricky way to make it an expression. Would probably be cleaner with 
`const mergedProps = {... rest, ...dialogProps }` and then `{...mergedProps}`